packages/grafana-data/src/transformations/transformers/calculateField.ts

```diff
@@ -235,16 +252,21 @@ export function getNameFromOptions(options: CalculateFieldTransformerOptions) {
     return options.alias;
   }
 
-  if (options.mode === CalculateFieldMode.BinaryOperation) {
-    const { binary } = options;
-    return `${binary?.left ?? ''} ${binary?.operator ?? ''} ${binary?.right ?? ''}`;
-  }
-
-  if (options.mode === CalculateFieldMode.ReduceRow) {
-    const r = fieldReducers.getIfExists(options.reduce?.reducer);
-    if (r) {
-      return r.name;
+  switch (options.mode) {
+    case CalculateFieldMode.BinaryOperation: {
+      const { binary } = options;
+      return `${binary?.left ?? ''} ${binary?.operator ?? ''} ${binary?.right ?? ''}`;
     }
+    case CalculateFieldMode.ReduceRow:
+      {
+        const r = fieldReducers.getIfExists(options.reduce?.reducer);
+        if (r) {
+          return r.name;
+        }
+      }
+      break;
```

Comment:
It's a JS thing. We don't need one in BinaryOperation because if it matches that case there is always a return. In ReduceRow we only hit the return if the reducer exists. In JS cases fall through by default, so you have to break or it will do the Index case

Reply:
its a js thing 🤣 -- yup switch in go has explicit fallthough and the opposite in js

Reply:
JS inherited a lot from C syntax, so this is actually a C-ism :stuck_out_tongue_winking_eye: